### PR TITLE
compilers/nim: do not modify hardening flags

### LINKED ISSUE #243759: buildNimPackage does not respect hardening settings
### Describe the bug

Setting `hardeningEnable = ["pie"];` or `hardeningDisable = ["all"];` in a `nimPackages.buildNimPackage`-based package results in the same output binary (partial RELRO, no PIE, as per `checksec`).

### Steps To Reproduce
`nix build github:eclairevoyant/pacclean` (which includes `hardeningEnable = ["pie"];`)

### Expected behavior
Respect hardening settings

### Screenshots
n/a

### Additional context
The only way currently to pass flags is via `nimFlags`, e.g. for enabling PIE, one must use

```nim
nimFlags = [
  "--passC:'-fPIE'"
  "--passL:'-fPIE -pie'"
];
```

But this is a rather clunky workaround and [is incorrectly documented](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/nim.section.md?plain=1#L76).

### Notify maintainers
@ehmry

### Metadata
n/a

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

Hardening can set -O𝑛 flags that contradict $CC flags added by the Nim compiler but this should be resolved later in the wrapper script.

Fix #243759

###### Things done

Nitter NixOS tests passes.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ehmry** [comment]
The `nimmm` package broke by the hardening adjustment, so I made some package updates to fix it and run more tests.

File: pkgs/development/nim-packages/ws/default.nix [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+buildNimPackage (final: {
+  pname = "ws";
+  version = "0.5.0";
+  src = fetchFromGitHub {
+    owner = "treeform";
+    repo = "ws";
+    rev = final.version;
+    hash = "sha256-3wVi6CjMTjcc5MJEozJN6W3TuYDb53w2MDCsv6lMH0k=";
+  };
+  preCheck = "rm tests/test_ws.nim tests/test_timeout.nim";
```
Comment (SuperSandro2000) [inline pkgs/development/nim-packages/ws/default.nix:12]:
```suggestion
  preCheck = ''
    rm tests/test_ws.nim tests/test_timeout.nim
  '';
```
preCheck is normally expected to be a multiline string

File: pkgs/development/nim-packages/ws/default.nix [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  meta = final.src.meta // {
```
Comment (SuperSandro2000) [inline pkgs/development/nim-packages/ws/default.nix:13]: Please copy the meta that src provides. It makes it a lot easier for other people to discover the homepage.

Also those packages do not seem to have meta.maintainer which is not allowed according to the contributing guide.

**Comment by tomcur** [comment]
Result of `nixpkgs-review pr 243803` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>hottext</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>nimPackages.eris</li>
    <li>nimPackages.eris.bin</li>
    <li>nimPackages.pixie</li>
    <li>nimPackages.regex</li>
  </ul>
</details>
<details>
  <summary>59 packages built:</summary>
  <ul>
    <li>mosdepth</li>
    <li>nim (nimPackages.nim)</li>
    <li>nimPackages.asciigraph</li>
    <li>nimPackages.asynctools</li>
    <li>nimPackages.base32</li>
    <li>nimPackages.base45</li>
    <li>nimPackages.bumpy</li>
    <li>nimPackages.c2nim</li>
    <li>nimPackages.cbor</li>
    <li>nimPackages.chroma</li>
    <li>nimPackages.coap</li>
    <li>nimPackages.flatty</li>
    <li>nimPackages.freedesktop_org</li>
    <li>nimPackages.getdns</li>
    <li>nimPackages.hts-nim</li>
    <li>nimPackages.illwill</li>
    <li>nimPackages.illwillwidgets</li>
    <li>nimPackages.jsonschema</li>
    <li>nimPackages.jsony</li>
    <li>nimPackages.nimSHA2</li>
    <li>nimPackages.nim_builder</li>
    <li>nimPackages.nimraylib-now</li>
    <li>nimPackages.nimsimd</li>
    <li>nimPackages.noise</li>
    <li>nimPackages.npeg</li>
    <li>nimPackages.parsetoml</li>
    <li>nimPackages.preserves</li>
    <li>nimPackages.rocksdb</li>
    <li>nimPackages.safeset</li>
    <li>nimPackages.sass</li>
    <li>nimPackages.sdl2</li>
    <li>nimPackages.segmentation</li>
    <li>nimPackages.snappy</li>
    <li>nimPackages.spry</li>
    <li>nimPackages.spryvm</li>
    <li>nimPackages.stew</li>
    <li>nimPackages.syndicate</li>
    <li>nimPackages.taps</li>
    <li>nimPackages.tempfile</li>
    <li>nimPackages.tkrzw</li>
    <li>nimPackages.ui</li>
    <li>nimPackages.unicodedb</li>
    <li>nimPackages.unicodeplus</li>
    <li>nimPackages.vmath</li>
    <li>nimPackages.ws</li>
    <li>nimPackages.x11</li>
    <li>nimPackages.zippy</li>
    <li>nimdow</li>
    <li>nimlsp</li>
    <li>nimmm</li>
    <li>nitch</li>
    <li>nitter</li>
    <li>nrpl</li>
    <li>promexplorer</li>
    <li>snekim</li>
    <li>swaycwd</li>
    <li>tridactyl-native</li>
    <li>ttop</li>
    <li>vimPlugins.fruzzy</li>
  </ul>
</details>

File: pkgs/development/nim-packages/regex/default.nix [hunk 3]
```diff
@@ -1,8 +1,16 @@
...
-  sha256 = "13gn0qhnxz07474kv94br5qlac9j8pz2555fk83538fiq83vgbm5";
-}
+buildNimPackage (final: {
+  pname = "regex";
+  version = "0.20.2";
+  src = fetchFromGitHub {
+    owner = "nitely";
+    repo = "nim-regex";
+    rev = "v${final.version}";
+    hash = "sha256-QD7X1QTAKHYa2+QiYjrQq74CnEafwq/RsshlW2yZqmI=";
```
Comment (tomcur) [inline pkgs/development/nim-packages/regex/default.nix:10]:
```suggestion
    hash = "sha256-VViDf0uD6bj9WOO827NRbLUt+PgBPEmz/A/DDRCrHpc=";
```

**Review by tomcur — CHANGES_REQUESTED** [review]

**Review by marcusramberg — APPROVED** [review]

**Comment by marcusramberg** [comment]
Seems eris is still failing actually. 
```
2 packages failed to build:
nimPackages.eris nimPackages.eris.bin
```

**Pull request merged**